### DISCUSSION TITLE
No LocalWorkers started when scheduler is launched in daemon mode.

### DISCUSSION THREAD

**Discussion by sorenarchibald** [apache/airflow#39385] [General]
### Apache Airflow version

2.9.0

### What happened?

LocalExecutor workers aren't started when launching the airflow scheduler in daemon mode.  I get this when running `apache/airflow:latest` docker image.  Additionally, the Web UI says the scheduler isn't running and Jobs aren't started.

### What you think should happen instead?

I would expect the same scheduler behavior whether launched as a deamon or not.

### How to reproduce

[compose.yml.txt](https://github.com/apache/airflow/files/15190709/compose.yml.txt)
1. Run the attached docker compose file.
2. login to UI and note the message that says the scheduler isn't running.
3. login to web container, as root.
4. run `apt-get update && apt-get install -y procps`
5. run `ps -ef` and note that none of the LocalExecutor workers are running.

I can workaround this issue by starting the webserver first and the scheduler last -- not in daemon mode.

### Operating System

PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"

### Versions of Apache Airflow Providers

N/A

### Deployment

Other Docker-based deployment

### Deployment details

Docker Compose version v2.26.1-desktop.1
Also observed run AWS  Elastic Container Service.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by RNHTTR** [comment]
The documented pattern for running Airflow in Docker can be found in the Airflow guide [Running Airflow in Docker](https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html). The docker compose file is a bit more extensive than the one you've shared. 

Can you try following that guide? You can run everything in detached mode with `docker compose up -d`.

**Comment by sorenarchibald** [comment]
Thanks Ryan.  I've worked around this issue by starting the scheduler last.  I'm actually using AWS ECS and only made the compose file to debug the issue.  I thought I would submit a bug as it did waste a couple hours of my time... and I thought it was time I try and get more involved with an open source project.  Again: thanks for your help.

**Reply by RNHTTR** [reply]
Happy to help :) I converted this to a discussion because I don't think there's actually a bug here. Again, I'd like to highlight the docs on [Running Airflow in Docker](https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html). Keep in mind that this is still not recommended for production.

**Comment by Taragolis** [comment]
Run Airflow services in daemon it is useful only on case if there is not possible for some reason to use external daemons (e.g. systemd) or containers.

In case of containers it's best practice to separate areas of concern by using one service per container.
It is much easier to maintain and restart due to the failure, e.g. if webserver goes down you don't have to restarts scheduler and vice versa.

> I'm actually using AWS ECS and only made the compose file to debug the issue.

Even if you use cloud based container service the principal pretty the same, 1 service = 1 task.
